go/vt/tabletserver/query_executor.go

```diff
@@ -296,14 +296,11 @@ func (qre *QueryExecutor) execDDL() (*sqltypes.Result, error) {
 	}
 
 	if qre.transactionID != 0 {
-		conn, err := qre.tsv.te.txPool.Get(qre.transactionID, "DDL implicit commit")
-		if err != nil {
-			return nil, err
-		}
-		err = qre.tsv.te.txPool.LocalCommit(qre.ctx, conn, qre.tsv.messager)
+		err := qre.tsv.te.txPool.Commit(qre.ctx, qre.transactionID, qre.tsv.messager)
```

Comment:
I don't think you can rely on `execAsTransaction` for this code path because it starts its own transaction. You have to perform your own `qre.execSQL` here. So, the flow should be:
```
txPool.Get
defer conn.Recycle
qre.execSQL
conn.BeginAgain (this is a method of TxConn)
return (skip execAsTransaction)
```

This way, the current TxConn retains its transaction id and it remains in the active pool. So, the subsequent commit from the app will be honored.